src/te_checks.c

```diff
@@ -611,39 +615,108 @@ struct check_result *check_declaration_interface_nameclash(__attribute__((unused
 	return NULL;
 }
 
-struct check_result *check_unknown_permission_macro(__attribute__((unused)) const struct check_data
-						    *data,
-						    const struct policy_node
-						    *node)
+struct check_result *check_unknown_permission(__attribute__((unused)) const struct check_data
+					      *data,
+					      const struct policy_node
+					      *node)
 {
-	static unsigned int permmacros_count = (unsigned int)-1;
-
-	if (permmacros_count == (unsigned int)-1) {
-		permmacros_count = permmacros_map_count();
+	static bool notified = false;
+	// ignore if no permission or permission macro have been parsed
+	if (permmacros_map_count() == 0) {
+		if (!notified) {
+			printf("%sNote%s: Check E-007 is not performed because no permission macro has been parsed.\n",
```

Comment:
Any particular reason to have this logic in the check rather than at enablement time?

I'm wondering about a general model where if a check needs particular resources to be available, we disable it while registering checks and notify at that point, instead of the first time it encounters it.

Reply:
I'll try to refactor